modules/light-clients/07-tendermint/types/store.go

```diff
@@ -94,3 +124,133 @@ func GetProcessedTime(clientStore sdk.KVStore, height exported.Height) (uint64,
+	height := binary.BigEndian.Uint64(heightBytes)
+	return clienttypes.NewHeight(revision, height)
+}
+
+func IterateConsensusStateAscending(clientStore sdk.KVStore, cb func(height exported.Height) (stop bool)) error {
+	iterator := sdk.KVStorePrefixIterator(clientStore, []byte(KeyIterateConsensusStatePrefix))
+	defer iterator.Close()
+
+	for ; iterator.Valid(); iterator.Next() {
+		iterKey := iterator.Key()
+		height := GetHeightFromIterationKey(iterKey)
+		if cb(height) {
+			return nil
+		}
+	}
+	return nil
+}
+
+// GetNextConsensusState returns the lowest consensus state that is larger than the given height.
+// The Iterator returns a storetypes.Iterator which iterates from start (inclusive) to end (exclusive).
+// Thus, to get the next consensus state, we must first call iterator.Next() and then get the value.
+func GetNextConsensusState(clientStore sdk.KVStore, cdc codec.BinaryMarshaler, height exported.Height) (*ConsensusState, bool) {
+	iterateStore := prefix.NewStore(clientStore, []byte(KeyIterateConsensusStatePrefix))
+	iterator := iterateStore.Iterator(bigEndianHeightBytes(height), nil)
+	defer iterator.Close()
+	// ignore the consensus state at current height and get next height
+	iterator.Next()
+	if !iterator.Valid() {
+		return nil, false
+	}
+
+	csKey := iterator.Value()
+	bz := clientStore.Get(csKey)
+	if bz == nil {
+		return nil, false
+	}
+
+	consensusStateI, err := clienttypes.UnmarshalConsensusState(cdc, bz)
+	if err != nil {
+		return nil, false
+	}
+
+	consensusState, ok := consensusStateI.(*ConsensusState)
+	if !ok {
+		return nil, false
+	}
+
+	return consensusState, true
+}
+
+// GetPreviousConsensusState returns the highest consensus state that is lower than the given height.
+// The Iterator returns a storetypes.Iterator which iterates from the end (exclusive) to start (inclusive).
+// Thus to get previous consensus state we call iterator.Value() immediately.
+func GetPreviousConsensusState(clientStore sdk.KVStore, cdc codec.BinaryMarshaler, height exported.Height) (*ConsensusState, bool) {
+	iterateStore := prefix.NewStore(clientStore, []byte(KeyIterateConsensusStatePrefix))
+	iterator := iterateStore.ReverseIterator(nil, bigEndianHeightBytes(height))
+	defer iterator.Close()
+
+	if !iterator.Valid() {
+		return nil, false
+	}
+
+	csKey := iterator.Value()
+	bz := clientStore.Get(csKey)
+	if bz == nil {
+		return nil, false
+	}
+
+	consensusStateI, err := clienttypes.UnmarshalConsensusState(cdc, bz)
+	if err != nil {
+		return nil, false
+	}
+
+	consensusState, ok := consensusStateI.(*ConsensusState)
+	if !ok {
+		return nil, false
+	}
+
+	return consensusState, true
+}
+
+func bigEndianHeightBytes(height exported.Height) []byte {
+	revisionBytes := make([]byte, 8)
+	heightBytes := make([]byte, 8)
+	binary.BigEndian.PutUint64(revisionBytes, height.GetRevisionNumber())
+	binary.BigEndian.PutUint64(heightBytes, height.GetRevisionHeight())
+	return append(revisionBytes, heightBytes...)
+}
```

Comment:
I believe it is more efficient to create one 16 byte slice. Appending the two slices causes a new slice to be allocated in order to double the capacity. 

I ran some benchmarks to verify.
Before:
```
Benchmark1-32    	 9852082	       110.4 ns/op	      40 B/op	       3 allocs/op
```
After:
```
Benchmark1-32    	18250534	        71.37 ns/op	      32 B/op	       2 allocs/op
```

Note, the first number is the amount of times the benchmark ran the for loop. As we can see allocating a 16 byte array is quite a bit faster. Almost twice as fast.

Here is the code I updated it to:
```go
func bigEndianHeightBytes(height exported.Height) []byte {
    heightBytes := make([]byte, 16) 
    binary.BigEndian.PutUint64(heightBytes[0:8], height.GetRevisionNumber())
    binary.BigEndian.PutUint64(heightBytes[8:], height.GetRevisionHeight())
    return heightBytes
}
```

Here is the benchmark I ran:

```go
package types_test

import (
    "testing"

    clienttypes "github.com/cosmos/ibc-go/modules/core/02-client/types"
    "github.com/cosmos/ibc-go/modules/light-clients/07-tendermint/types"
)

func Benchmark1(b *testing.B) {
    benchmarkBigEndian(b)
}

func benchmarkBigEndian(b *testing.B) {
    b.ReportAllocs()

    for i := 0; i < b.N; i++ {

        height := clienttypes.NewHeight(uint64(i), uint64(i))

        types.BigEndianHeightBytes(height)
    }   
}
```